src/OrdinaryDiffEq.jl

```diff
@@ -192,90 +193,118 @@ using DocStringExtensions
-
-        AutoVern9(Rodas5P()), AutoVern9(Rodas5P(autodiff=false)),
-        AutoVern9(Rodas5P(chunk_size = 1)),
-        AutoVern9(Rodas5P(chunk_size = Val{1}())),
-
-        AutoVern7(Rodas4()), AutoVern7(Rodas4(autodiff=false)),
-        #AutoVern7(Rodas4(chunk_size = 1)),
-        #AutoVern7(Rodas4(chunk_size = Val{1}())),
-
-        #AutoVern7(Rodas5P()), AutoVern7(Rodas5P(autodiff=false)),
-        #AutoVern7(Rodas5P(chunk_size = 1)),
-        #AutoVern7(Rodas5P(chunk_size = Val{1}())),
-
-        AutoVern7(TRBDF2()), AutoVern7(TRBDF2(autodiff=false)),
-        #AutoVern7(TRBDF2(chunk_size = 1)),
-        #AutoVern7(TRBDF2(chunk_size = Val{1}())),
-      ]
+SnoopPrecompile.@precompile_all_calls begin
+  function lorenz(du, u, p, t)
+    du[1] = 10.0(u[2] - u[1])
+    du[2] = u[1] * (28.0 - u[3]) - u[2]
+    du[3] = u[1] * u[2] - (8 / 3) * u[3]
+  end
 
-      prob_list = [
-        ODEProblem(lorenz,[1.0;0.0;0.0],(0.0,1.0))
-        ODEProblem{true,false}(lorenz,[1.0;0.0;0.0],(0.0,1.0))
-        ODEProblem{true,false}(lorenz,[1.0;0.0;0.0],(0.0,1.0),Float64[])
-        ODEProblem(lorenz_oop,[1.0;0.0;0.0],(0.0,1.0))
-        #ODEProblem{false,false}(lorenz_oop,[1.0;0.0;0.0],(0.0,1.0))
-        #ODEProblem{false,false}(lorenz_oop,[1.0;0.0;0.0],(0.0,1.0),Float64[])
-      ]
+  function lorenz_oop(u, p, t)
+    [10.0(u[2] - u[1]), u[1] * (28.0 - u[3]) - u[2], u[1] * u[2] - (8 / 3) * u[3]]
+  end
+  nonstiff_solver_options =
+    ["BS3" => BS3(), "Tsit5" => Tsit5(), "Vern7" => Vern7(), "Vern9" => Vern9()]
+  stiff_solver_options = [
+    ("Rosenbrock23", "true;true;true;true", Rosenbrock23),
+    ("Rodas4", "true;true;false;false", Rodas4),
+    ("Rodas5", "true;true;false;false", Rodas5),
+    ("Rodas5P", "true;true;true;true", Rodas5P),
+    ("TRBDF2", "true;true;false;false", TRBDF2),
+    ("KenCarp4", "true;true;false;false", KenCarp4),
+    ("QNDF", "true;true;false;false", QNDF),
+  ]
+  autoswitch_solver_options = [
+    ("AutoTsit5Rosenbrock23", "true;true;true;true", AutoTsit5, Rosenbrock23),
+    ("AutoTsit5TRBDF2", "true;true;false;false", AutoTsit5, Rosenbrock23),
+    ("AutoTsit5Rodas5P", "false;false;false;false", AutoTsit5, Rodas5P),
+    ("AutoVern9KenCarp47", "true;true;false;false", AutoVern9, KenCarp47),
+    ("AutoVern9Rodas5", "true;true;true;true", AutoVern9, Rodas5),
+    ("AutoVern9Rodas5P", "true;true;true;true", AutoVern9, Rodas5P),
+    ("AutoVern7Rodas4", "true;true;false;false", AutoVern7, Rodas4),
+    ("AutoVern7Rodas5P", "false;false;false;false", AutoVern7, Rodas5P),
+    ("AutoVern7TRBDF2", "true;true;false;false", AutoVern7, TRBDF2),
+  ]
+
+
+  solver_list = []
+  for (solvername, solver) in nonstiff_solver_options
+    if Preferences.@load_preference(solvername, "true") == "true"
+      push!(solver_list, solver)
+    end
+  end
+  for (solvername, default, solvertype) in stiff_solver_options
+    options = split(Preferences.@load_preference(solvername, default), ';')
```

Comment:
A bit simpler here would also be
```suggestion
    options = Preferences.@load_preference(solvername, default)
```
and use defaults of e.g. `[true, false, false, true]`.